tests/test_other.py

```diff
@@ -9888,16 +9888,27 @@ def test_emranlib(self):
     output = run_process([shared.LLVM_NM, '--print-armap', 'liba.a'], stdout=PIPE).stdout
     self.assertContained('Archive map', output)
 
+  def test_preprocess_stdin(self):
+    create_test_file('temp.h', '#include <string>')
+    outputStdin = run_process([PYTHON, EMCC, '-x', 'c++', '-dM', '-E', '-'], input="#include <string>", stdout=PIPE).stdout
+    outputFile = run_process([PYTHON, EMCC, '-x', 'c++', '-dM', '-E', 'temp.h'], stdout=PIPE).stdout
+    self.assertTextDataIdentical(outputStdin, outputFile)
+
+  def test_compile_stdin(self):
+    with open(path_from_root('tests', 'hello_world.cpp')) as f:
+      run_process([PYTHON, EMCC, '-x', 'c++', '-'], input=f.read())
+    self.assertContained('hello, world!', run_js('a.out.js'))
+
   def test_pthread_stub(self):
     # Verify that programs containing pthread code can still be compiled even
     # without enabling threads.  This is possible becase we link in
     # libpthread_stub.a
     create_test_file('pthread.c', '''
-#include <pthread.h>
+      #include <pthread.h>
 
-int main() {
-  pthread_atfork(NULL, NULL, NULL);
-  return 0;
-}
-''')
+      int main() {
+      pthread_atfork(NULL, NULL, NULL);
+      return 0;
+      }
+      ''')
```

Comment:
Can you revert this part?

Reply:
Do you want me to remove it entirely, was due to a merge from incoming.

Reply:
Yes, if it's not relevant to this PR, then it doesn't need to be here (even if it showed up in the merge commit).

Reply:
I'm not sure how that change got in there, but its showing up as diff from origin/upstream.  If you are having trouble figuring out git I can lang this change for you without this part. 

One way to revert part of change like this is to do `git checkout -p origin/incoming <filename>` then say "yes" to the part you want to revert, then `git commit`.

Reply:
I have reverted it, thanks for the tip. Sorry for the inconvenience, still learning.